include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp

```diff
@@ -53,27 +53,56 @@ namespace pressio{ namespace ops{
 
 //----------------------------------------------------------------------
 //  overloads for computing: MV = a * MV + b * MV1
-// where MV is an tpetra multivector wrapper
+// where MV is an tpetra multivector
 //----------------------------------------------------------------------
 
-template<typename T, typename a_t, class b_t>
+template<typename T, typename T1, typename a_t, typename b_t>
 ::pressio::mpl::enable_if_t<
-  ::pressio::is_multi_vector_tpetra_block<T>::value
+  // rank-1 update common constraints
+     ::pressio::Traits<T>::rank == 2
+  && ::pressio::Traits<T1>::rank == 2
+  // TPL/container specific
+  && ::pressio::is_multi_vector_tpetra_block<T>::value
+  && ::pressio::is_multi_vector_tpetra_block<T1>::value
+  // scalar compatibility
+  && ::pressio::all_have_traits_and_same_scalar<T, T1>::value
+  && (std::is_floating_point<typename ::pressio::Traits<T>::scalar_type>::value
+   || std::is_integral<typename ::pressio::Traits<T>::scalar_type>::value)
+  && std::is_convertible<a_t, typename ::pressio::Traits<T>::scalar_type>::value
+  && std::is_convertible<b_t, typename ::pressio::Traits<T>::scalar_type>::value
   >
 update(T & mv, const a_t &a,
-       const T & mv1, const b_t &b)
+       const T1 & mv1, const b_t &b)
 {
-  mv.update(b, mv1, a);
+  assert(::pressio::ops::extent(mv, 0) == ::pressio::ops::extent(mv1, 0));
+  assert(::pressio::ops::extent(mv, 1) == ::pressio::ops::extent(mv1, 1));
+
+  using scalar_t = typename ::pressio::Traits<T>::scalar_type;
+  const scalar_t a_{a};
+  const scalar_t b_{b};
+
+  mv.update(b_, mv1, a_);
 }
 
-template<typename T, typename scalar_t>
+template<typename T, typename T1, typename b_t>
 ::pressio::mpl::enable_if_t<
-  ::pressio::is_multi_vector_tpetra_block<T>::value
+  // rank-1 update common constraints
+     ::pressio::Traits<T>::rank == 2
+  && ::pressio::Traits<T1>::rank == 2
+  // TPL/container specific
+  && ::pressio::is_multi_vector_tpetra_block<T>::value
+  && ::pressio::is_multi_vector_tpetra_block<T1>::value
+  // scalar compatibility
+  && ::pressio::all_have_traits_and_same_scalar<T, T1>::value
+  && (std::is_floating_point<typename ::pressio::Traits<T>::scalar_type>::value
+   || std::is_integral<typename ::pressio::Traits<T>::scalar_type>::value)
+  && std::is_convertible<b_t, typename ::pressio::Traits<T>::scalar_type>::value
   >
-update(T & mv, const T & mv1, const scalar_t & b)
+update(T & mv, const T1 & mv1, const b_t & b)
 {
+  using scalar_t = typename ::pressio::Traits<T>::scalar_type;
   constexpr auto zero = ::pressio::utils::Constants<scalar_t>::zero();
-  mv.update(b, mv1, zero);
+  pressio::ops::update(mv, zero, mv1, b);
```

Comment:
same comment as in the tpetra PR

Reply:
fixed on f8e4aa5361a8b1203b86da9b130d67ad612bb53f